app/workers/document_list_export_worker.rb

```diff
@@ -15,8 +15,13 @@ def perform(filter_options, user_id)
 
   def send_mail(csv_filename, user, filter)
     MailNotifications.document_list(csv_filename, user.email, filter.page_title).deliver_now
-  rescue Notifications::Client::BadRequestError
-    # unrecoverable when running asynchronously
+  rescue Notifications::Client::BadRequestError => e
+    if !%w[integration staging].include?(ENV["SENTRY_CURRENT_ENV"]) || e.message !~ /team-only API key/
+      # in production we care about all errors
+      # in staging and integration the team-only error is unrecoverable when running asynchronously
+      # (team-only error is unrecoverable in production too, but almost certainly impossible)
+      raise
```

Comment:
We should probably re-raise the exception we caught, rather than raise with no arguments.

```suggestion
raise e
```

Reply:
Actually [if you call `raise` without any arguments it will re-raise the last error](https://stackoverflow.com/a/23771227), which I think can help in avoiding an extra unnecessary line in the stacktrace.

Reply:
Oh, cool! Apologies, I should have checked first.